codegen/attribute_analyzer.go

```diff
@@ -0,0 +1,191 @@
+package codegen
+
+import (
+	"goa.design/goa/expr"
+)
+
+type (
+	// AttributeAnalyzer analyzes an attribute and its sub-attributes.
```

Comment:
This is a big interface which is not idiomatic. I wonder if we couldn't just use structs instead but need to understand better the interactions.